x-pack/platform/plugins/shared/ml/public/application/components/import_export_jobs/import_jobs_flyout/jobs_import_service.ts

```diff
@@ -155,29 +253,122 @@ export class JobImportService {
+        }
+      });
+    }
+
+    if (type === 'anomaly-detector') {
+      // exclude jobs with missing filters
+      const validJobsForDatafeedCheck = (jobs as ImportedAdJob[]).filter(
+        (j) => !commonJobs.find((cj) => cj.jobId === j.job.job_id)?.filters?.length
+      );
+
+      const datafeedResults = await this.validateDatafeeds(validJobsForDatafeedCheck);
+      datafeedResults.forEach((result) => {
+        datafeedValidations.set(result.jobId, result);
+      });
+    }
+
+    commonJobs.forEach(({ jobId, filters = [], destIndex }) => {
       const missingFilters = filters.filter((i) => existingFilters.has(i) === false);
+      const indicesErrors = sourceIndicesErrors.get(jobId);
 
-      if (missingIndices.length === 0 && missingFilters.length === 0) {
+      if (missingFilters.length === 0 && !indicesErrors) {
         tempJobs.push({
           jobId,
           ...(type === 'data-frame-analytics' ? { destIndex } : {}),
         });
       } else {
         skippedJobs.push({
           jobId,
-          missingIndices,
-          missingFilters,
+          ...(missingFilters.length > 0 ? { missingFilters } : {}),
+          ...(indicesErrors ? { sourceIndicesErrors: indicesErrors } : {}),
         });
       }
     });
 
     return {
       jobs: tempJobs,
       skippedJobs,
+      datafeedValidations,
     };
   }
+
+  public async validateDatafeeds(jobs: ImportedAdJob[]): Promise<DatafeedValidationResult[]> {
+    const results = await Promise.all(
+      jobs.map(async ({ job, datafeed }) => {
+        try {
+          const combinedJob: CombinedJob = {
+            ...job,
+            datafeed_config: datafeed,
+          };
+
+          const response: DatafeedValidationResponse = await this.validateDatafeedPreview({
+            job: combinedJob,
+          });
+
+          if (response.error) {
+            return {
+              jobId: job.job_id,
+              hasWarning: true,
+              warningMessage: i18n.translate('xpack.ml.jobsList.datafeedPreviewValidationFailed', {
+                defaultMessage: `Unable to validate datafeed preview. Reason: {reason}`,
+                values: {
+                  reason: extractErrorMessage(response.error),
+                },
+              }),
+            };
+          }
+
+          if (!response.valid) {
+            return {
+              jobId: job.job_id,
+              hasWarning: true,
+              warningMessage: i18n.translate('xpack.ml.jobsList.datafeedPreviewFailed', {
+                defaultMessage: 'Datafeed preview failed. This job may not run correctly.',
+              }),
+            };
+          }
+
+          if (!response.documentsFound) {
+            return {
+              jobId: job.job_id,
+              hasWarning: true,
+              warningMessage: i18n.translate('xpack.ml.jobsList.datafeedPreviewNoData', {
+                defaultMessage: 'Datafeed preview returned no data. This job will not run.',
```

Comment:
I think the text `This job will not run.` is incorrect. The job will run, it just won't produce any results because it can't find data.
This could use a similar message as above:
`Datafeed preview returned no data. This job may not run correctly.`